src/mds/MDSRank.cc

```diff
@@ -3115,7 +3115,7 @@ void MDSRankDispatcher::evict_clients(
   dout(20) << __func__ << " matched " << victims.size() << " sessions" << dendl;
 
   if (victims.empty()) {
-    on_finish(0, {}, outbl);
+    on_finish(-CEPHFS_EINVAL, "Invalid value", outbl);
```

Comment:
If we pass `0` instead of `error exit status` we won't be able to catch it, as it won't raise command failed error . And we also won't be able to test it.
We will see this error: ```Traceback (most recent call last):
2024-06-07 23:47:14,537.537 INFO:__main__:  File "/home/neerajpratapsingh/ceph/qa/tasks/cephfs/test_misc.py", line 596, in test
2024-06-07 23:47:14,537.537 INFO:__main__:    getattr(self, t)(cmd)
2024-06-07 23:47:14,537.537 INFO:__main__:  File "/home/neerajpratapsingh/ceph/qa/tasks/cephfs/test_misc.py", line 557, in _evict_with_invalid_id
2024-06-07 23:47:14,537.537 INFO:__main__:    with self.assertRaises(CommandFailedError) as ce:
2024-06-07 23:47:14,537.537 INFO:__main__:AssertionError: CommandFailedError not raised```

Any other exit status?

Reply:
As passing an `id` to evict which doesn't match to any hosts should be finished with some exit status to know  that id doesn't exist along with exit message.

Reply:
Let's use ESRCH then. We want to give an error different from EINVAL if the filter is valid but no session matches.

Reply:
Updated!! PTAL